tests/test_orient_sdp.py

```diff
@@ -0,0 +1,193 @@
+import numpy as np
+import pytest
+
+from aspire.abinitio import CommonlineSDP
+from aspire.nufft import backend_available
+from aspire.source import Simulation
+from aspire.utils import (
+    Rotation,
+    get_aligned_rotations,
+    register_rotations,
+    rots_to_clmatrix,
+)
+from aspire.volume import AsymmetricVolume
+
+RESOLUTION = [
+    32,
+    33,
+]
+
+OFFSETS = [
+    None,  # Defaults to random offsets.
+    0,
+]
+
+DTYPES = [
+    np.float32,
+    pytest.param(np.float64, marks=pytest.mark.expensive),
+]
+
+
+@pytest.fixture(params=RESOLUTION, ids=lambda x: f"resolution={x}")
+def resolution(request):
+    return request.param
+
+
+@pytest.fixture(params=OFFSETS, ids=lambda x: f"offsets={x}")
+def offsets(request):
+    return request.param
+
+
+@pytest.fixture(params=DTYPES, ids=lambda x: f"dtype={x}")
+def dtype(request):
+    return request.param
+
+
+@pytest.fixture
+def src_orient_est_fixture(resolution, offsets, dtype):
+    """Fixture for simulation source and orientation estimation object."""
+    src = Simulation(
+        n=50,
+        L=resolution,
+        vols=AsymmetricVolume(L=resolution, C=1, K=100, seed=0).generate(),
+        offsets=offsets,
+        amplitudes=1,
+        seed=0,
+    )
+
+    max_shift = 1 / src.L  # sets max_shift to 1 pixel for 0 offsets.
+    shift_step = 1
+    if src.offsets.all() != 0:
```

Comment:
I don't think this does what you think it does…

Reply:
It does function in the way that I expect for this test, but I can imagine a case where this doesn't do what I'm intending here. Namely, if there happens to be a zero anywhere in our random offsets.

Let me clean this up to have the intended result in all cases.

Reply:
Sort of inverted the logic here so we check for when all offsets are zero. Let me know what you think.

Reply:
Also, using `np.all()` properly now.